src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -36,6 +36,37 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+    /**
+     * Returns a string of the type of attributes that is unique to a {@code person} that already exist
+     * in the address book.
+     */
+    public String getNonUniqueAttributeType(Person toCheck) {
+        requireNonNull(toCheck);
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < internalList.size(); i++) {
+            Person otherPerson = internalList.get(i);
+            if (otherPerson.isSamePerson(toCheck)) {
+                if (otherPerson.getPhone().equals(toCheck.getPhone()) && !stringBuilder.toString().contains("Phone")) {
+                    stringBuilder.append("Phone");
+                }
+                if (otherPerson.getMatriculationNumber().equals(toCheck.getMatriculationNumber())
```

Comment:
This method is a bit long! might want to consider abstracting out the checks if you have the time 😄